### ISSUE TITLE
Run syntax migration in topo order

### ISSUE THREAD

**Issue by CJ-Wright** [regro/cf-scripts#218]
@justcalamari would it be possible to do the syntax migration in topo (or pseudo-topo) order? I don't think we need to have the syntax changes depend on parent syntax changes, but it would be nice to do the first generation first.

Do we have a pseudo-topo order for the graph?

**Comment by justcalamari** [comment]
The `cyclic_topological_sort` function needs a source node to start from, since topological sort will normally start from a node with no incoming edges, which may not be possible in the conda-forge graph with its cycles. If there is a node from which every other node is reachable, we should be able to run `cyclic_topological_sort` starting from that.

Since we only care about syntax migration in topo order for now, we can run starting from toolchain and then any unreachable nodes in the order networkx returns. But if we want to run in topo order for all migrators, we have to identify a package (or perhaps group of packages) to start from.

**Comment by CJ-Wright** [comment]
Hmm ok. I think the toolchain fix would work for now. We may need to add something to the migrators that allows for sorting the graph in different ways? (Esp. if we can't find a universal first generation package)

@isuruf @mariusvniekerk @msarahan @scopatz any idea if there is a first generation package we can use in the topo sort?

**Comment by scopatz** [comment]
Maybe make or perl or Python

**Comment by CJ-Wright** [comment]
@justcalamari we just need to pick something from the top cycle, right?

**Comment by isuruf** [comment]
what about `compiler_stub` (can't remember the exact name that the parser changes {{compiler}} jinja to)

**Comment by justcalamari** [comment]
The graph isn't connected.
~~~
c99-to-c89
cfortran-cfitsio
containerpilot
cyan
dirent
dmd
git-lfs
go-spatial
inkscape
libunistring
vs2015_runtime', 'miktex
nanoflann
pstdint
pv
seqan-library
su-exec
utfcpp
flock
datamash
liquidprompt
libassuan
libksba
npth
luajit
bisonpp
conda-forge-build-setup
vs2008_express_vc_python_patch
osx-pocl-opencl
dataspyre
nose-parameterized
lammps
libsigsegv
airflow-split
racket
coreutils
~~~
Aren't connected to the rest of the graph. Some of these might be due to errors in the bot.

**Comment by justcalamari** [comment]
@isuruf `compiler_stub` will probably work after the syntax migrations finish, but that won't work yet since those updates haven't finished.

**Comment by scopatz** [comment]
Also, there are likely some packages that will never have any parents, even with the compiler stub

**Comment by justcalamari** [comment]
Solved by #225
